File: app/src/main/java/com/sports/oscaracademy/adapters/dashBoard_adapter.java
package com.sports.oscaracademy.adapters;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.sports.oscaracademy.data.DashBoardData;
import com.sports.oscaracademy.databinding.SingleDashboardRcvBinding;
import com.sports.oscaracademy.homeActivities.FeesPayment;
import com.sports.oscaracademy.homeActivities.PayAndPlay;
import com.sports.oscaracademy.homeActivities.Students;
import com.sports.oscaracademy.homeActivities.attendance;
import com.sports.oscaracademy.homeActivities.news_feeds;
import com.sports.oscaracademy.homeActivities.schedule.Schedule;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;

public class dashBoard_adapter extends RecyclerView.Adapter<dashBoard_adapter.holder> {
    private final ArrayList<DashBoardData> data;
    Context context;
    private final SharedPreferences pref;
    private final String isStudent;
    private final String role;

    public dashBoard_adapter(ArrayList<DashBoardData> data, Context context) {
        this.data = data;
        this.context = context;
        pref = context.getSharedPreferences("tokenFile", Context.MODE_PRIVATE);
        isStudent = pref.getString("isStudent", "false");
        role = pref.getString("userType", "-1");
    }

    @NonNull
    @Override
    public holder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        SingleDashboardRcvBinding binding = SingleDashboardRcvBinding.inflate(LayoutInflater.from(parent.getContext()),parent,false);
        return new holder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull dashBoard_adapter.holder holder, int position) {
       DashBoardData d =  data.get(position);
       holder.binding.setUserModelRCV(d);
        Log.d("TAG", "onBindViewHolder: " + d.getFieldname());
       holder.binding.executePendingBindings();
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    public class holder extends RecyclerView.ViewHolder{
        private final SingleDashboardRcvBinding binding;
        public holder(@NonNull @NotNull SingleDashboardRcvBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
            binding.cardView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    nextScreen(String.valueOf(binding.textView.getText()));
                }
            });
        }
        public void nextScreen(String text){
            Intent i;
            switch (text.trim()) {
                case "Fees & Payments":
                    if (isStudent.equals("true")) {
                        context.startActivity(new Intent(context, FeesPayment.class));
                    } else {
                        Toast.makeText(context, "Only Available For Enrolled Students", Toast.LENGTH_SHORT).show();
                    }
                    break;
                case "Attendance":
                    if (isStudent.equals("true") || role.equals("-2")) {
                        context.startActivity(new Intent(context, attendance.class));
                    } else {
                        Toast.makeText(context, "Only Available For Enrolled Students", Toast.LENGTH_SHORT).show();
                    }
                    break;
                case "News Feeds": //for both admin and students
                    context.startActivity(new Intent(context, news_feeds.class));
                    break;
                case "Schedule":
                    if (isStudent.equals("true")) {
                        context.startActivity(new Intent(context, Schedule.class));
                    } else {
                        Toast.makeText(context, "Only Available For Enrolled Students", Toast.LENGTH_SHORT).show();
                    }
                    break;
                case "Students":
                    i = new Intent(context, Students.class);
                    i.putExtra("catcher", "0");
                    context.startActivity(i);
                    break;

                case "Pay & Play":
                    context.startActivity(new Intent(context, PayAndPlay.class));
                    break;

                case "Add Student":
                    i = new Intent(context, Students.class);
                    i.putExtra("catcher", "1");
                    context.startActivity(i);
                    break;

                case "Update Schedule":
                    break;

            }
        }
    }

}

File: adminpanel/src/main/java/com/geek/adminpanel/adapter/DashboardAdapter.kt
package com.geek.adminpanel.adapter

import android.content.Context
import android.util.Log
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.navigation.NavController
import androidx.navigation.Navigation
import androidx.recyclerview.widget.RecyclerView
import com.geek.adminpanel.R
import com.geek.adminpanel.databinding.SingleDashboardRcvBinding
import java.util.*

class DashboardAdapter(
    val data: ArrayList<DashboardData>?,
    private val context: Context,
    private val navController: NavController,
    private val clickListener: OnClickInterface
) :
    RecyclerView.Adapter<DashboardAdapter.DashViewHolder>() {
    private var pref = context.getSharedPreferences("tokenFile", Context.MODE_PRIVATE)
    private var isStudent: String? = null
    private var role: String? = null

    init {
        isStudent = pref.getString("isStudent", "false")
        role = pref.getString("userType", "-1")
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DashViewHolder {
        val binding: SingleDashboardRcvBinding =
            SingleDashboardRcvBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return DashViewHolder(binding, clickListener)
    }

    override fun onBindViewHolder(holder: DashViewHolder, position: Int) {
        val d: DashboardData = data!![position]
        val imageID = d.imageID
        holder.binding.image.setImageResource(imageID)
        holder.binding.textView.text = d.fieldName
        Log.d("TAG", "onBindViewHolder: " + d.fieldName)
        holder.binding.executePendingBindings()
    }

    override fun getItemCount(): Int {
        return data!!.size
    }

    class DashViewHolder(binding: SingleDashboardRcvBinding, val clickListener: OnClickInterface) :
        RecyclerView.ViewHolder(binding.root) {
        val binding: SingleDashboardRcvBinding = binding
        fun nextScreen(text: String) {
            when (text.trim()) {
                "Change Schedule" -> {
                    Navigation.findNavController(binding.root)
                        .navigate(R.id.action_adminDashboard_to_changeSchedule)
                }
                "Slots Timings" -> {
                    Navigation.findNavController(binding.root)
                        .navigate(R.id.action_adminDashboard_to_slotTimings)
                }
                "Fees History" -> {
                    Navigation.findNavController(binding.root)
                        .navigate(R.id.action_adminDashboard_to_feesPayments)
                }
                "Academy Contacts" -> {
                    clickListener.OnRcvItemSelected("Academy Contacts")
                }
                "Edit Court Price" -> {
                    clickListener.OnRcvItemSelected("Edit Court Price")
                }
                "User Feedback" -> {
                    Navigation.findNavController(binding.root)
                        .navigate(R.id.action_adminDashboard_to_userFeedbacks)
                }
                "Admin/Coaches" -> {
                    Navigation.findNavController(binding.root)
                        .navigate(R.id.action_adminDashboard_to_adminCoach)
                }
            }
        }

        init {
            binding.cardView.setOnClickListener {
                nextScreen(
                    java.lang.String.valueOf(
                        binding.textView.text
                    )
                )
            }
        }
    }

}

interface OnClickInterface {
    fun OnRcvItemSelected(name: String)
}

data class DashboardData(val imageID: Int, val fieldName: String)